### DISCUSSION TITLE
New Table Key System

### DISCUSSION THREAD

**Discussion by insomniacj93** [Budibase/budibase#9285] [Help]
Hello guys,

Hope christmas was good and you'll had a great new year!

I've got a question about the new side panel system with click row selection.

I already have a screen with a table, I was hoping to basicly build myself the whole side panel interface with the row selection instead of relying on CRUD.

![Screenshot_20230109_032226](https://user-images.githubusercontent.com/107686004/211347068-967647e0-d77e-4615-9450-41b872edc3cf.png) (Table)
![Screenshot_20230109_032252](https://user-images.githubusercontent.com/107686004/211347082-b6a4bdb3-dd94-4641-9d3f-5d2b85eddd44.png) (Side panel)

How are these Keys generated?
Can we use the same key in a different screen or can it be as simple as key1 and create our own keys?

I haven't found any documentation yet, but if there is i would very much appreciate it!

Thank you in advance!

**Comment by melohagan** [comment]
Hey @insomniacj93 

If you don't want the full CRUD functionality, you can do the following:

1. Untick `Show button above table` to remove the Create option
![Screenshot 2023-01-10 at 13 58 57](https://user-images.githubusercontent.com/101575380/211570611-2d23451b-f771-4655-b2cb-eedbe838f25c.png)

2. Eject the table block
3. In the Details side panel, untick `Show save button` and `Show delete button`
![Screenshot 2023-01-10 at 14 00 02](https://user-images.githubusercontent.com/101575380/211570891-235b9361-710b-4dda-870a-b3bd3f16ef05.png)

You could also change the Form block type to **View** which would also disable the form fields. 

---

> How are these Keys generated?

They are randomly generated when a table block is created through an auto-generated screen. 

> Can we use the same key in a different screen or can it be as simple as key1 and create our own keys?

You can see that the *Persist this value* checkbox is not ticked, which means that state cannot be reference in other screens by default. You could use your own App State variable, or even use [URL variables](https://docs.budibase.com/docs/url-parameters)
